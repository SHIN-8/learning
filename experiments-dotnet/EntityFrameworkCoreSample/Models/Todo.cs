using System.Collections.Generic;

namespace EntityFrameworkCoreSample.Models
{
    /// <summary>
    /// Todoエンティティ
    /// </summary>
    public class Todo
    {
       /// <summary>
       /// Id
       /// </summary>
       /// <value></value>
       public int Id { get; set; }
       
       /// <summary>
       /// Description
       /// </summary>
       /// <value></value>
       public string Description { get; set; }


       /// <summary>
       /// isActive
       /// </summary>
       /// <value></value>
       public bool isActive { get; set; }

    }
}