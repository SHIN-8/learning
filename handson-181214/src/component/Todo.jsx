import * as React from 'react'
import * as ReactDOM from 'react-dom'

//---------------------------
//component
//---------------------------
export function Todo(props) {
    let key = 0
    return (<div style={{ textAlign: 'left', margin: 12 }}>
        <button
            onClick={props.onClickIncrement} >
            TODOを追加する
        </button><br />
        <table style={{ align: 'center' }}>
            <thead>
                <tr>
                    <th>内容</th>
                    <th>操作</th>
                </tr>
            </thead>
            <tbody>
                {
                    props.todo.map((element, index, array) => {
                        return <tr>
                            <td>
                                <input value={element} onChange={props.onChange.bind(this, index)} key={key++} />
                            </td>
                            <td>
                                <button onClick={props.onClickDecrement.bind(this, index)} key={key++} >
                                    削除する
                    </button>
                            </td>
                        </tr>
                    })
                }
            </tbody>
        </table>
    </div>)

}