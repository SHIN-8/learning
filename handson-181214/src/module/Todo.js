import * as React from 'react'
import * as ReactRedux from 'react-redux'
import { createAction, handleActions } from 'redux-actions'

//-----------------------
//Model
//-----------------------
export const InitialTodoStore = {
  todo: [''],
}

//-----------------------
//Action
//-----------------------
export const OnChangeActionName = "Todo/onChange"
export const OnClickInclimentActionName = "Todo/onClickIncrement"
export const OnClickDeclimentActionName = "Todo/onClickDecliment"


//-----------------------
//ActionCreator
//-----------------------

export const onChange = createAction(OnChangeActionName)
export const onIncrement = createAction(OnClickInclimentActionName)
export const onDecliment = createAction(OnClickDeclimentActionName)


//-----------------------
//Reducer
//-----------------------
export const TodoReducer = handleActions({
  [OnChangeActionName]: (state, action) => {
    const todo = state.todo.slice(0)
    const { index, value } = action.payload
    console.log(todo)
    todo[index] = value
    return { todo: todo }

  },

  [OnClickInclimentActionName]: (state, action) => {
    const todo = state.todo.slice(0)
    todo.push('')
    console.log(todo)
    return { todo: todo }
  },

  [OnClickDeclimentActionName]: (state, action) => {
    const todo = state.todo.slice(0)
    const { index } = action.payload
    todo.splice(index, 1)
    console.log(todo)
    return { todo: todo }

  },


}, InitialTodoStore)


export default TodoReducer

