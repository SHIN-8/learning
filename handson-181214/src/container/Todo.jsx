import * as React from 'react'
import * as ReactDom from 'react-dom'
import { Todo } from '../component/Todo'
import { connect } from 'react-redux'
import * as TodoModule from '../module/Todo'

function mapStateToProps(state) {
    return {
        ...state.todo
    };
}

function mapDispatchToAciton(dispatch) {
    return {
        onChange(index, e) {
            dispatch(TodoModule.onChange({ index: index, value: e.target.value }))
        },
        onClickIncrement() {
            dispatch(TodoModule.onIncrement())
        },
        onClickDecrement(index) {
            dispatch(TodoModule.onDecliment({ index: index }))
        },
    }
}

export default connect(
    mapStateToProps,
    mapDispatchToAciton
)(Todo);