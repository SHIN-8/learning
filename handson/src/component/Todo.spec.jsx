import * as React from "react"
import * as ReactDom from "react-dom"
import * as Enzyme from "enzyme"
import { Todo } from "./Todo"

const Adapter = require("enzyme-adapter-react-16")
Enzyme.configure({ adapter: new Adapter() })

const EmptyHandler = () => {

}

export default describe("Todo Module", () => {
    beforeEach(() => {

    })

    afterEach(() => {

    })

    it("add", () => {
        const component = Enzyme.shallow(<Todo
            todo={[]}
            onChange={EmptyHandler}
            onDecliment={EmptyHandler}
            onIncrement={EmptyHandler}
        />)
        expect(component).toMatchSnapshot()
    })

    it("add", () => {
        const component = Enzyme.shallow(<Todo
            todo={[]}
            onChange={EmptyHandler}
            onDecliment={EmptyHandler}
            onIncrement={EmptyHandler}
        />)
        expect(component).toMatchSnapshot()
    })

    it("add", () => {
        const component = Enzyme.shallow(<Todo
            todo={[]}
            onChange={EmptyHandler}
            onDecliment={EmptyHandler}
            onIncrement={EmptyHandler}
        />)
        expect(component).toMatchSnapshot()
    })
});