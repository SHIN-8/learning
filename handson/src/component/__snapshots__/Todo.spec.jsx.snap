// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Module add 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todo
    onChange={[Function]}
    onDecliment={[Function]}
    onIncrement={[Function]}
    todo={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button>
          TODOを追加する
        </button>,
        <br />,
        <table
          style={
            Object {
              "align": "center",
            }
          }
        >
          <thead>
            <tr>
              <th>
                内容
              </th>
              <th>
                操作
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
      ],
      "style": Object {
        "margin": 12,
        "textAlign": "left",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TODOを追加する",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "TODOを追加する",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                  内容
                </th>
                <th>
                  操作
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
          "style": Object {
            "align": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  内容
                </th>
                <th>
                  操作
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    内容
                  </th>,
                  <th>
                    操作
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "内容",
                  },
                  "ref": null,
                  "rendered": "内容",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "操作",
                  },
                  "ref": null,
                  "rendered": "操作",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button>
            TODOを追加する
          </button>,
          <br />,
          <table
            style={
              Object {
                "align": "center",
              }
            }
          >
            <thead>
              <tr>
                <th>
                  内容
                </th>
                <th>
                  操作
                </th>
              </tr>
            </thead>
            <tbody />
          </table>,
        ],
        "style": Object {
          "margin": 12,
          "textAlign": "left",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TODOを追加する",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "TODOを追加する",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                  <th>
                    内容
                  </th>
                  <th>
                    操作
                  </th>
                </tr>
              </thead>,
              <tbody />,
            ],
            "style": Object {
              "align": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    内容
                  </th>
                  <th>
                    操作
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      内容
                    </th>,
                    <th>
                      操作
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "内容",
                    },
                    "ref": null,
                    "rendered": "内容",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "操作",
                    },
                    "ref": null,
                    "rendered": "操作",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Todo Module add 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todo
    onChange={[Function]}
    onDecliment={[Function]}
    onIncrement={[Function]}
    todo={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button>
          TODOを追加する
        </button>,
        <br />,
        <table
          style={
            Object {
              "align": "center",
            }
          }
        >
          <thead>
            <tr>
              <th>
                内容
              </th>
              <th>
                操作
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
      ],
      "style": Object {
        "margin": 12,
        "textAlign": "left",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TODOを追加する",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "TODOを追加する",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                  内容
                </th>
                <th>
                  操作
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
          "style": Object {
            "align": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  内容
                </th>
                <th>
                  操作
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    内容
                  </th>,
                  <th>
                    操作
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "内容",
                  },
                  "ref": null,
                  "rendered": "内容",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "操作",
                  },
                  "ref": null,
                  "rendered": "操作",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button>
            TODOを追加する
          </button>,
          <br />,
          <table
            style={
              Object {
                "align": "center",
              }
            }
          >
            <thead>
              <tr>
                <th>
                  内容
                </th>
                <th>
                  操作
                </th>
              </tr>
            </thead>
            <tbody />
          </table>,
        ],
        "style": Object {
          "margin": 12,
          "textAlign": "left",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TODOを追加する",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "TODOを追加する",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                  <th>
                    内容
                  </th>
                  <th>
                    操作
                  </th>
                </tr>
              </thead>,
              <tbody />,
            ],
            "style": Object {
              "align": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    内容
                  </th>
                  <th>
                    操作
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      内容
                    </th>,
                    <th>
                      操作
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "内容",
                    },
                    "ref": null,
                    "rendered": "内容",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "操作",
                    },
                    "ref": null,
                    "rendered": "操作",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Todo Module add 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todo
    onChange={[Function]}
    onDecliment={[Function]}
    onIncrement={[Function]}
    todo={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button>
          TODOを追加する
        </button>,
        <br />,
        <table
          style={
            Object {
              "align": "center",
            }
          }
        >
          <thead>
            <tr>
              <th>
                内容
              </th>
              <th>
                操作
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
      ],
      "style": Object {
        "margin": 12,
        "textAlign": "left",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TODOを追加する",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "TODOを追加する",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                  内容
                </th>
                <th>
                  操作
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
          "style": Object {
            "align": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  内容
                </th>
                <th>
                  操作
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    内容
                  </th>,
                  <th>
                    操作
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "内容",
                  },
                  "ref": null,
                  "rendered": "内容",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "操作",
                  },
                  "ref": null,
                  "rendered": "操作",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button>
            TODOを追加する
          </button>,
          <br />,
          <table
            style={
              Object {
                "align": "center",
              }
            }
          >
            <thead>
              <tr>
                <th>
                  内容
                </th>
                <th>
                  操作
                </th>
              </tr>
            </thead>
            <tbody />
          </table>,
        ],
        "style": Object {
          "margin": 12,
          "textAlign": "left",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TODOを追加する",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "TODOを追加する",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                  <th>
                    内容
                  </th>
                  <th>
                    操作
                  </th>
                </tr>
              </thead>,
              <tbody />,
            ],
            "style": Object {
              "align": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    内容
                  </th>
                  <th>
                    操作
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      内容
                    </th>,
                    <th>
                      操作
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "内容",
                    },
                    "ref": null,
                    "rendered": "内容",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "操作",
                    },
                    "ref": null,
                    "rendered": "操作",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
